adoc/admin-monitoring-health-checks.adoc

```diff
@@ -0,0 +1,373 @@
+
+There's also the https://kubernetes.io/docs/tasks/debug-application-cluster/monitor-node-health/[Monitor Node Health] check. This is a `DaemonSet` that runs on every node, and reports to the `apiserver` back as `NodeCondition` and `Events`.
+
+== Service/Application Health Checks
+
+If the deployed services contain a health endpoint, or if they contain an endpoint that can be used to determine if the service is up, you can use `livenessProbes` and/or `readinessProbes`.
+
+.Health check endpoints vs. functional endpoints
+[NOTE]
+====
+A proper health check is always preferred if designed correctly.
+
+Despite the fact that any endpoint could potentially be used to infer if your application is up, a specific health endpoint in your application is preferred.
+Such an endpoint will only respond affirmatively when all your setup code on the server has finished and the application is running in a desired state.
+====
+
+The `livenessProbes` and `readinessProbes` share configuration options and probe types.
+
+initialDelaySeconds::
+Number of seconds to wait before performing the very first liveness probe.
+
+periodSeconds::
+Number of seconds that the kubelet should wait between liveness probes.
+
+successThreshold::
+Number of minimum consecutive successes for the probe to be considered successful (Default: 1).
+
+failureThreshold::
+Number of times this probe is allowed to fail in order to assume that the service is not responding (Default: 3).
+
+timeoutSeconds::
+Number of seconds after which the probe times out (Default: 1).
+
+There are different options for the `livenessProbes` to check:
+
+Command::
+A command executed within a container; a reture code of 0 means success. All other return codes mean failure.
+
+TCP::
+If a TCP connection can be established is considered success.
+
+HTTP::
+Any HTTP response between `200` and `400` indicates success.
+
+=== livenessProbe
+
+https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/[livenessProbes] are used to detect running but misbehaving pods/a service that might be running (the process didn't die), but that is not responding as expected.
+
+Probes are executed by each `kubelet` against the pods that define them and that are running in that specific node.
+
+When a `livenessProbe` fails, {kube} will automatically restart the pod and increase the `RESTARTS` count for that pod.
+
+These probes will be executed every `periodSeconds` starting from `initialDelaySeconds`.
+
+=== readinessProbe
+
+https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/#define-readiness-probes[readinessProbes] are used to wait for processes that take some time to start. Despite the container running, it might be performing some time consuming initializatoin operations. During this time, you don't want {kube} to route traffic to that specific pod; also, you don't want that container to be restarted because it will appear unresponsive.
```

Comment:
```suggestion
https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/#define-readiness-probes[readinessProbes] are used to wait for processes that take some time to start. Despite the container running, it might be performing some time consuming initialization operations. During this time, you don't want {kube} to route traffic to that specific pod; also, you don't want that container to be restarted because it will appear as unresponsive.
```